### PR TITLE
Use paths action for filtering jobs instead of separate ignore workflows

### PR DESCRIPTION
**Pull request by chainchad**

### CODE REVIEW

**Comment by chainchad** [comment]
This is what we do for the integration-tests.yml and solidity*.yml workflows.

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -11,35 +6,46 @@ on:
...
           fetch-depth: 0
       - uses: actions/setup-go@v3
+        if: github.event.schedule != '' || needs.changes.outputs.src == 'true'
         with:
           go-version-file: 'go.mod'
       - name: golangci-lint
+        # any scheduled run or push/pulls with relevant changed paths, but skip PRs from release branches
+        if: |
+          github.event.schedule != '' ||
+          (needs.changes.outputs.src == 'true' && (github.event_name == 'push' || (github.event_name == 'pull_request' && !startsWith(github.head_ref, 'release/')))
```
Comment (HenryNguyen5) [inline .github/workflows/golangci-lint.yml:47]:
```suggestion
          (needs.changes.outputs.src == 'true' && (github.event_name == 'push' || github.event_name == 'pull_request' )
```

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -11,35 +6,46 @@ on:
       - staging
       - trying
       - rollup
-    paths:
-      - '**.go'
-      - '**/go.mod'
-      - '**/go.sum'
-      - '.golangci.yml'
   pull_request:
```
[the remark this answers is not in the document]
Comment (HenryNguyen5) [reply .github/workflows/golangci-lint.yml:9]: Could we do this instead?

**Review by HenryNguyen5 — CHANGES_REQUESTED** [review]

**Review by jmank88 — DISMISSED** [review]

File: .github/workflows/golangci-lint.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (chainchad) [reply .github/workflows/golangci-lint.yml:9]: Good idea. There was a formatting issue in this suggestion so I made the change in 045f78bf317b8c6c78c3d594b64863b55146089a.

**Review by jmank88 — APPROVED** [review]

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (chainchad) [reply .github/workflows/golangci-lint.yml]: This simplifies things nicely

**Review by HenryNguyen5 — DISMISSED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**